spec/scenarios/dunning/campaign_v1_spec.rb

```diff
@@ -0,0 +1,232 @@
+          external_id: external_subscription_id,
+          plan_code: plan.code
+        }
+      )
+      perform_billing
+
+      expect(webhooks_sent.map { _1["webhook_type"] }).to eq(%w[
+        subscription.started
+        invoice.created
+        invoice.generated
+        invoice.payment_failure
+      ])
+      invoice = customer.invoices.sole
+      expect(invoice.payment_status).to eq("failed")
+      expect(invoice.payment_due_date).to eq(Date.new(2025, 1, 3))
+    end
+
+    # The day after payment_due_date, the invoice should be marked as overdue
+    travel_to(DateTime.new(2025, 1, 4, 13)) do
+      perform_overdue_balance_update
+
+      invoice = customer.invoices.sole
+      expect(invoice).to be_payment_overdue
+      expect(customer.overdue_balance_cents).to eq(149_00)
+
+      # Performing dunning has no effect because the threshold is 150 and we have only 149 overdue
+      perform_dunning
+      expect(customer.payment_requests.count).to eq(0)
+    end
+
+    # Create a one-off invoice to reach the threshold
+    travel_to(DateTime.new(2025, 1, 4, 18)) do
+      create_one_off_invoice(customer, [addon], units: 3)
+      perform_all_enqueued_jobs
+
+      oneoff = customer.invoices.one_off.sole
+      expect(oneoff.payment_status).to eq("failed")
+      expect(oneoff.payment_due_date).to eq(Date.new(2025, 1, 6))
+    end
+
+    travel_to(DateTime.new(2025, 1, 7, 10)) do
+      perform_overdue_balance_update
+      expect(customer.invoices.one_off.sole).to be_payment_overdue
+
+      expect(ActionMailer::Base.deliveries.count).to eq(0)
+      perform_dunning
+      # NOTE: Email is sent twice: first synchronously after Stripe response, and then when the webhook is received
+      expect(ActionMailer::Base.deliveries.count).to eq(2)
+      expect(ActionMailer::Base.deliveries.map(&:subject)).to all eq "Your overdue balance from JC AI"
+
+      mail = ActionMailer::Base.deliveries.last
+      expect(mail.subject).to eq "Your overdue balance from JC AI"
+
+      pr = customer.payment_requests.sole
+      expect(pr.amount_cents).to eq(155_00)
+
+      # TODO Simulate webhook payment failed received
+    end
+
+    # The next 2 days nothing happens
+    [DateTime.new(2025, 1, 8, 10), DateTime.new(2025, 1, 9, 10)].each do |date|
+      travel_to(date) do
+        perform_overdue_balance_update
+        perform_dunning
+        expect(ActionMailer::Base.deliveries.count).to eq(2) # nothing new
+      end
+    end
+
+    # The next day nothing happens
+    travel_to(DateTime.new(2025, 1, 10, 10)) do
+      perform_overdue_balance_update
+      perform_dunning
+      expect(ActionMailer::Base.deliveries.count).to eq(4)
+
+      expect(customer.payment_requests.reload.map(&:amount_cents)).to all eq 155_00
+    end
+
+    # This is over
+    travel_to(DateTime.new(2025, 1, 13, 13)) do
+      perform_overdue_balance_update
+      perform_dunning
+      expect(ActionMailer::Base.deliveries.count).to eq(4) # Nothing new
+    end
```

Comment:
it may be worth to change the comment to mention the dunning campaign is over and no new attempt is processed after 2 days from last because campaign is set to max attempts 2... it will make a lot easier to understand the test without having to check the objects, etc.